### PR TITLE
Change subqueries with pk filters to have realtime semantics

### PR DESCRIPTION
**Pull request by mfussenegger**
This removes the ESGetTask and instead adds a PKLookupPhase & Operation
which ties into our execution infrastructure.

This way the "Get" specific code no longer has to take care of ORDER BY,
LIMIT, OFFSET. This is handled by other operators in the planner, and by
projections in the executor.

Furthermore it eliminates a couple of restrictions regarding the use of `_version` in the where clause. 

From a performance perspective this change seems to be an improvement as well. Running https://github.com/crate/crate-benchmarks/blob/master/py/specs/select.toml yielded:

```
Q: select * from uservisits where "sourceIP" = '25.193.131.52'
C: 1
  76.82% mean difference. Likely significant
             V1  →     V2
  mean:    0.944 →   0.420
  stdev:   0.554 →   0.265
  max:     7.649 →   5.848
  min:     0.372 →   0.230
```

### CODE REVIEW

**Comment by mfussenegger** [comment]
retest this please

File: sql/src/test/groovy/io/crate/planner/InsertPlannerTest.java [hunk 1]
```diff
@@ -261,6 +280,52 @@ public void testInsertFromSubQueryESGet() throws Exception {
         assertThat(projection.partitionedBySymbols().isEmpty(), is(true));
     }
 
+    private ClusterState buildClusterStateWithUsersIndex(ClusterState prevState) {
+        /*
+          So far we relied on dummy getRouting implementations on {@link io.crate.metadata.table.TestingTableInfo}
+          so that we could test the plan creation. The actual clusterState never contained any meta data
+          (= no tables; no shard routing table)
+
+          This is adds the users table to the cluster state (though without mapping)
```
Comment (matriv) [inline sql/src/test/groovy/io/crate/planner/InsertPlannerTest.java]: remove `is`

File: sql/src/main/java/io/crate/operation/PKLookupOperation.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public final class PKLookupOperation {
+
+    private final IndicesService indicesService;
+
+    public PKLookupOperation(IndicesService indicesService) {
```
[the remark this answers is not in the document]
Comment (mfussenegger) [reply sql/src/main/java/io/crate/operation/PKLookupOperation.java]: Don't really see the benefit in doing so

[the remark this answers is not in the document]
Comment (matriv) [reply sql/src/main/java/io/crate/operation/PKLookupOperation.java]: ok, not important, just wanted to know if missed something and it wasn't possible

**Pull request merged**